src/client/pps.go

```diff
@@ -329,116 +383,88 @@ func (c APIClient) ListJobFilterF(pipelineName string, inputCommit []*pfs.Commit
 	}
 }
 
-// FlushJob calls f with all the jobs which were triggered by commits.
-// If toPipelines is non-nil then only the jobs between commits and those
-// pipelines in the DAG will be returned.
-func (c APIClient) FlushJob(commits []*pfs.Commit, toPipelines []string, f func(*pps.JobInfo) error) error {
-	req := &pps.FlushJobRequest{
-		Commits: commits,
-	}
-	for _, pipeline := range toPipelines {
-		req.ToPipelines = append(req.ToPipelines, NewPipeline(pipeline))
-	}
-	client, err := c.PpsAPIClient.FlushJob(c.Ctx(), req)
+// SubscribeJob calls the given callback with each open job in the given
+// pipeline until canceled.
+func (c APIClient) SubscribeJob(pipelineName string, includePipelineInfo bool, cb func(*pps.JobInfo) error) error {
+	client, err := c.PpsAPIClient.SubscribeJob(
+		c.Ctx(),
+		&pps.SubscribeJobRequest{
+			Pipeline: NewPipeline(pipelineName),
+			Full:     includePipelineInfo,
+		})
 	if err != nil {
 		return grpcutil.ScrubGRPC(err)
 	}
 	for {
-		jobInfo, err := client.Recv()
-		if err != nil {
-			if errors.Is(err, io.EOF) {
-				return nil
-			}
+		ji, err := client.Recv()
+		if errors.Is(err, io.EOF) {
+			return nil
+		} else if err != nil {
 			return grpcutil.ScrubGRPC(err)
 		}
-		if err := f(jobInfo); err != nil {
+		if err := cb(ji); err != nil {
+			if errors.Is(err, errutil.ErrBreak) {
+				return nil
+			}
 			return err
 		}
 	}
 }
 
-// FlushJobAll returns all the jobs which were triggered by commits.
-// If toPipelines is non-nil then only the jobs between commits and those
-// pipelines in the DAG will be returned.
-func (c APIClient) FlushJobAll(commits []*pfs.Commit, toPipelines []string) ([]*pps.JobInfo, error) {
-	var result []*pps.JobInfo
-	if err := c.FlushJob(commits, toPipelines, func(ji *pps.JobInfo) error {
-		result = append(result, ji)
-		return nil
-	}); err != nil {
-		return nil, err
-	}
-	return result, nil
-}
-
 // DeleteJob deletes a job.
-func (c APIClient) DeleteJob(jobID string) error {
+func (c APIClient) DeleteJob(pipelineName string, jobID string) error {
```

Comment:
This should probably just take a `pps.Job`.

Reply:
I'm planning to do a pass soon to change this across all our functions